spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/autoconfig/PropertyBasedBaggageFields.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.autoconfig;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import brave.baggage.BaggageField;
+
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.env.Environment;
+
+import static java.util.Arrays.asList;
+
+/**
+ * Wire up property-based {@linkplain BaggageField baggage fields}, so that they appear as
+ * if they were defined one-by-one. This allows users to contribute fields and also for us
+ * to use the list of them above.
+ */
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnProperty(value = "spring.sleuth.enabled", matchIfMissing = true)
+public class PropertyBasedBaggageFields implements BeanFactoryPostProcessor {
```

Comment:
@dsyer @wilkinsona either of you know how to do what I'm trying to do here? MultipleHopsIntegrationTests fails because this can't read any of the properties. Most tests pass though.

Basically, I want to programmatically register these constants so that other things can base on them. They need no information except a property to do that, and I spent several hours trying things and failed I guess.

any other way to accomplish same would be great!

Reply:
sorry.. this was a test config problem. the application name was wrong.

anyway feedback welcome if you think there's a cleaner way to accomplish this.